### DISCUSSION TITLE
First limiting case issue

### DISCUSSION THREAD

**Discussion by Star-Ruby** [Part-II-Computational-Physics/.github#60] [Q&A]
The magnetisation graph doesn't go high or low enough for me in the first limiting case when I use a random lattice, but it works fine for the totally spin 1/-1 lattices. My code for evolving the magnetisation is
```
def evolve_and_compute_M(lattice,betaJ,betaMuH,avg_times):
    M = []
    for t in range(avg_times[-1]+1):
      if t in avg_times: M.append(compute_magnetisation(lattice))
      attempt_spin_flip(lattice,len(lattice),betaJ,betaMuH)
    return np.average(M)
```
![image](https://user-images.githubusercontent.com/90109705/223427358-f5b93418-daef-48ce-b771-db855d5be9e0.png)
![image](https://user-images.githubusercontent.com/90109705/223427663-cf9ed059-89b7-4fb5-a0b2-57c436d89cc8.png)

**Comment by DLHvdH** [comment]
The code you posted look alright. Are you making sure that you start averaging at a high enough time step? If the system hasn't equilibrated yet, you might end up averaging in states with a zero average magnetisation.

**Reply by Star-Ruby** [reply]
What about averaging during these times? I get a smoother curve for the random lattice with the same kind of error. It should have thermalised by then? 
![image](https://user-images.githubusercontent.com/90109705/223525971-97aaf582-003b-4e03-abf4-6b6c0f3665b6.png)

**Reply by DLHvdH** [reply]
Hmmm 10 000 time steps should be enough. Did you check your computation of the magnetisation? Check if a lattice with only +1 spins indeed gives a magnetisation of M = 1. Likewise for a lattice with only -1 to give M = -1.

**Reply by Star-Ruby** [reply]
It returns the expected values.
![image](https://user-images.githubusercontent.com/90109705/223575180-5b82e7b8-d06f-45ba-8d78-f9fd3921f85a.png)
This is my function for computing the magnetisation, 
![image](https://user-images.githubusercontent.com/90109705/223575234-711acd74-abab-4a10-88ee-cd2f7902ac0c.png)
And my function for evolving it,
![image](https://user-images.githubusercontent.com/90109705/223575272-7d78062a-c15a-425d-84d0-c38c0adebe6c.png)
